core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -355,10 +355,10 @@ default void dropColumn(ConnectorSession session, ConnectorTableHandle tableHand
     default Optional<ConnectorNewTableLayout> getInsertLayout(ConnectorSession session, ConnectorTableHandle tableHandle)
     {
         ConnectorTableProperties properties = getTableProperties(session, tableHandle);
+        Map<ColumnHandle, String> columnNamesByHandle = getColumnHandles(session, tableHandle).entrySet().stream()
+                .collect(toMap(Map.Entry::getValue, Map.Entry::getKey));
         return properties.getTablePartitioning()
                 .map(partitioning -> {
-                    Map<ColumnHandle, String> columnNamesByHandle = getColumnHandles(session, tableHandle).entrySet().stream()
```

Comment:
it';s not a loop

Reply:
@findepi could you please advice better message?

Reply:
Unless I am wrong here, we're inside `Optional#map` call. 
Thus, extracting the call before makes the `getColumnHandles` being called unconditionally, also when not needed.

Reply:
@findepi, thank you!